tests/unit/plugins/callback/test_opentelemetry.py

```diff
@@ -91,3 +91,66 @@ def test_finish_task_without_a_host_match(self):
         self.assertEqual(host_data.uuid, 'include')
         self.assertEqual(host_data.name, 'include')
         self.assertEqual(host_data.status, 'ok')
+
+    def test_get_error_message_with_exception(self):
+        res_data = OrderedDict()
+        res_data['exception'] = 'my-exception'
+        res_data['msg'] = 'my-msg'
+
+        result = self.opentelemetry.get_error_message(res_data)
+        self.assertEqual(result, 'my-exception')
+
+    def test_get_error_message_without_exception(self):
+        res_data = OrderedDict()
+        res_data['msg'] = 'my-msg'
+
+        result = self.opentelemetry.get_error_message(res_data)
+        self.assertEqual(result, 'my-msg')
+
+    def test_get_error_message_without_exception_and_msg(self):
+        res_data = OrderedDict()
+
+        result = self.opentelemetry.get_error_message(res_data)
+        self.assertEqual(result, 'failed')
+
+    def test_enrich_error_message(self):
+        res_data = OrderedDict()
+        res_data['exception'] = 'my-exception'
+        res_data['msg'] = 'my-msg'
+        res_data['stderr'] = 'my-stderr'
+
+        result = self.opentelemetry.enrich_error_message(res_data)
+        self.assertEqual(result, 'message: "my-msg"\nexception: "my-exception"\nstderr: "my-stderr"')
+
+    def test_get_error_message_without_msg(self):
+        res_data = OrderedDict()
+        res_data['exception'] = 'my-exception'
+        res_data['stderr'] = 'my-stderr'
+
+        result = self.opentelemetry.enrich_error_message(res_data)
+        self.assertEqual(result, 'message: "failed"\nexception: "my-exception"\nstderr: "my-stderr"')
+
+    def test_get_error_message_without_msg(self):
+        res_data = OrderedDict()
+        res_data['msg'] = 'my-msg'
+        res_data['stderr'] = 'my-stderr'
+
+        result = self.opentelemetry.enrich_error_message(res_data)
+        self.assertEqual(result, 'message: "my-msg"\nexception: "None"\nstderr: "my-stderr"')
+
+    def test_get_error_message_without_stderr(self):
+        res_data = OrderedDict()
+        res_data['exception'] = 'my-exception'
+        res_data['msg'] = 'my-msg'
+
+        result = self.opentelemetry.enrich_error_message(res_data)
+        self.assertEqual(result, 'message: "my-msg"\nexception: "my-exception"\nstderr: "None"')
+
+    def test_get_error_message_with_multiple_line_stderr(self):
+        res_data = OrderedDict()
+        res_data['exception'] = 'my-exception'
+        res_data['msg'] = 'my-msg'
+        res_data['stderr'] = '\nmake[1]: Entering directory \n.No space left on device'
+
+        result = self.opentelemetry.enrich_error_message(res_data)
+        self.assertEqual(result, 'message: "my-msg"\nexception: "my-exception"\nstderr: "\nmake[1]: Entering directory \n.No space left on device"')
```

Comment:
Would you be able to condense these tests to two methods?
One for `get_error_message` and the other for `enrich_error_message` with each case run as a subtest.

Reply:
I refactored those tests, see https://github.com/ansible-collections/community.general/pull/3496/commits/13dbeae5d89b59c37074d535fedaaff3bfd73699

Thanks for the review :)

Reply:
I was thinking more along the lines of:

```python
def test_get_error_message(self):
        test_cases = (
            ('my-exception', 'my-msg', None, 'my-exception'),
            (None, 'my-msg', None, 'my-msg'),
            (None, None, None, 'failed'),
        )

        for tc in test_cases:
            result = self.opentelemetry.get_error_message(generate_test_data(tc[0], tc[1], tc[2]))
            self.assertEqual(result, tc[3])
```

There are better solutions, but anything that makes the tests more maintainable would be nice.

Reply:
See https://github.com/ansible-collections/community.general/pull/3496/commits/16c2d96b164ae764be7e143bdea7d66b54d33181

Thanks for the review